File: app/src/main/java/com/example/map/data/local/dao/FavoriteDocumentDao.java
package com.example.map.data.local.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.map.data.local.model.DocumentEntity;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Flowable;

@Dao
public interface FavoriteDocumentDao {
    @Query("SELECT * FROM documententity")
    Flowable<List<DocumentEntity>> getAll();

    @Query("SELECT * FROM documententity WHERE id =:id")
    Flowable<DocumentEntity> get(String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insert(DocumentEntity documentEntity);

    @Delete
    Completable delete(DocumentEntity documentEntity);

}

File: app/src/main/java/com/example/map/data/local/dao/FavoriteDocumentDao.kt
package com.example.map.data.local.dao

import androidx.room.*
import com.example.map.data.local.model.DocumentEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface FavoriteDocumentDao {
    @Query("SELECT * FROM documententity")
    fun getAll(): Flow<List<DocumentEntity>>

    @Query("SELECT * FROM documententity WHERE id =:id")
    fun get(id: String): Flow<DocumentEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(documentEntity: DocumentEntity)

    @Delete
    suspend fun delete(documentEntity: DocumentEntity)
}
